src/api/DataSource/index.ts

```diff
@@ -162,7 +163,7 @@ export class DataSourceAPI extends Request {
    */
   public createSourceConfig(this: Request<number>, params: SelectedDataSourceConfig) {
     return this.request({
-      errors: [UNAUTHORIZED, FORBIDDEN],
+      errors: [UNAUTHORIZED, FORBIDDEN, INVALID_PARAMETER],
```

Comment:
I was checking the server code, I see that when we create a source config we try to connect as well

https://github.com/Linkurious/linkurious-server/blob/b644710cf6f06fb060a1235dcb97df6c49fbfd42/server/services/data/index.ts#L502

and inside the connect function we reject with a `DATA_SOURCE_UNAVAILABLE` error if we fail to connect

https://github.com/Linkurious/linkurious-server/blob/b644710cf6f06fb060a1235dcb97df6c49fbfd42/server/services/data/dataSource.ts#L749

so should we add the `DATA_SOURCE_UNAVAILABLE` to the list of expected errors ?